worker/uniter/runner/jujuc/juju-log_test.go

```diff
@@ -21,35 +26,129 @@ var _ = gc.Suite(&JujuLogSuite{})
 
 func (s *JujuLogSuite) newJujuLogCommand(c *gc.C) cmd.Command {
 	ctx, _ := s.newHookContext(-1, "")
-	com, err := jujuc.NewCommand(ctx, cmdString("juju-log"))
+	cmd, err := jujuc.NewJujuLogCommand(ctx)
 	c.Assert(err, jc.ErrorIsNil)
-	return com
+	return cmd
+}
+
```

Comment:
I would pack more into this method - just pass the mock controller and have it provide all the constituents.

There is no need to pass a gc.C, as there is never a non-nil error kicking around.
```
func NewMockedJujuLogCommand(ctrl *gomock.MockController) (cmd.Command, JujuLogContext, JujuLogCommandLogger) {
        ctx := jujuc.NewMockJujuLogContext(ctrl)
        logger := logger := jujuc.NewMockJujuLogCommandLogger(ctrl)
        cmd := &JujuLogCommand{
		ctx: ctx,
		getLogger: func(s string) JujuLogCommandLogger {
			return logger
		},
	}

        return cmd, ctx, logger
}
```

expectLogf would need to receive the mocked logger instead of a mock controller. Maybe even pass the expected messages into the method above.

Reply:
This is changed in the new commit https://github.com/juju/juju/pull/8818/commits/946d44b04ec31e444974abfbb53f0a080a2ed999